ucan/src/ucan.rs

```diff
@@ -222,6 +171,46 @@ impl Ucan {
     pub fn version(&self) -> &str {
         &self.header.ucv
     }
+
+    fn try_from_token_string(ucan_token: &str) -> Result<Self> {
+        // better to create multiple iterators than collect, or clone.
+        let signed_data = ucan_token
+            .split('.')
+            .take(2)
+            .map(String::from)
+            .reduce(|l, r| format!("{}.{}", l, r))
+            .ok_or_else(|| anyhow!("Could not parse signed data from token string"))?;
+
+        let mut parts = ucan_token.split('.').map(|str| {
+            base64::decode_config(str, base64::URL_SAFE_NO_PAD).map_err(|error| anyhow!(error))
+        });
```

Comment:
Dropping this minor adjustment for consideration.

```suggestion
        let parts = ucan_token.split('.');
        let signed_data = parts.clone().take(2).collect::<Vec<_>>().join(".");
        let mut parts = parts
            .map(|str| base64::decode_config(str, base64::URL_SAFE_NO_PAD).map_err(|error| anyhow!(error)));
```

Reply:
@appcypher I had gone similarly on another go but was trying to avoid the clone and collect, as multiple iterators are typically more efficient than vec allocation here (from what I've read). Of course, going w/ collect solves for readability have splitting twice.

Reply:
reading online seems to prefer the current version, but I can go either way.

Reply:
I'll let @cdata decide, easy enough change. I tried some other funky things, but yeah.

Reply:
You are right. I think the existing version is fine.